### PR TITLE
Various improvements and small fixes of the MET uncertainty tools

### PR DESCRIPTION
**Pull request by mmarionncern**
This PR contains several fixes, improvements  and simplification of the MET uncertainty code and MET correction code. Also prepare the handling of the MVA MET uncertainties, planned for 76X.

Modified packages are :
- DataFormats/PatCandidates
- JetMETCorrections/Type1MET
- PhysicsTools/PatAlgos
- PhysicsTools/PatUtils

Code modifications :
- add the photon ES uncertainties in the pat::MET format, was missing before
- add automatic swich for JEC corrections betwee data and MC in the MET type1 code
- move to CHS jets and CHS JEC correctiosn for the type1 MET per default
- jet soft cleaning (electron/muon overlap removal) added per default in the uncertainty tool
- update of the XY MET shift correction (74X version, latest one)
- fix a bug in the XY MET shift correction computation
- update of the type0 correction (74X version, latest one)
- add shortcuts functions for the miniAOD production and re-production, with the minimal set of options needed by non-expert people. 
- add example config file to show how to reprocess the MET on top of miniAODs

Expected changes :
- central MET value will change a bit due to the switch to CHS, no strong effect expected
- JES/JER MET uncertainties can change a bit due to the jet cleaning
- XY and T0 changes will affect all METs with at least one of those corrections

### CODE REVIEW

**Comment by monttj** [comment]
please test

**Comment by monttj** [comment]
@mmarionncern 
The test failed with corrMET_pfMet_cfg.py file. The error message is below.  

An exception of category 'ProductNotFound' occurred while
   [0] Processing run: 1 lumi: 2 event: 101
   [1] Running path 'p'
   [2] Calling event method for module PFJetMETcorrInputProducer/'corrPfMetType1'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: reco::JetCorrector
Looking for module label: ak4PFCHSL1FastL2L3Corrector
Looking for productInstanceName:

**Comment by mmarionncern** [comment]
@monttj it is on my to do list, will be fixed tomorrow afternoon at the latest

**Comment by monttj** [comment]
please test

**Comment by mmarionncern** [comment]
Looks like a DAS error, not sure if it comes from that PR of from a temporary issue with DAS

**Comment by mariadalfonso** [comment]
Hi all, what is the status of this PR ? Maria

**Comment by mmarionncern** [comment]
@cmsbuild, @vadler, @monttj could you please review the PR ?

**Comment by monttj** [comment]
please test

**Comment by mmarionncern** [comment]
@cmsbuild branch has been rebased

**Comment by mmarionncern** [comment]
@cmsbuild @monttj : can the tests be triggered? Thanks

**Comment by smuzaffar** [comment]
please test

**Comment by gpetruc** [comment]
Hi,

According to DAS, all the needed files for these Matrix tests for these
runs are not at cern, i.e. it's not a glitch of the system and re-doing
"please test" won't help.

1003.0 uses `dataset=/MinimumBias/Run2012A-v1/RAW run=191226` whose
corresponding block
`/MinimumBias/Run2012A-v1/RAW#b0847904-85d5-11e1-a9bb-003048caaace` is only
at T0_CH_CERN and T1_US_FNAL (
https://cmsweb.cern.ch/das/request?input=site%20block%3D/MinimumBias/Run2012A-v1/RAW%23b0847904-85d5-11e1-a9bb-003048caaace&instance=prod/global&idx=0&limit=10
)

4.22 uses `/Cosmics/Run2011A-v1/RAW run=160960` whose block
/Cosmics/Run2011A-v1/RAW#d299346e-5384-11e0-9b51-00151755cb60
<https://cmsweb.cern.ch/das/request?input=block%3D%2FCosmics%2FRun2011A-v1%2FRAW%23d299346e-5384-11e0-9b51-00151755cb60&instance=prod%2Fglobal>
is at T0, CNAF and FNAL

1000.0, 1001.0 use `/MinimumBias/Run2011A-v1/RAW run=165121` whose block is
`/MinimumBias/Run2011A-v1/RAW#f6f3c7cc-7fbf-11e0-b519-003048caaace`, at T0,
FNAL, GRIF

I have made a Phedex request to bring them in the relval pool (
https://cmsweb.cern.ch/phedex/prod/Request::View?request=471555 )

Giovanni

**Comment by gpetruc** [comment]
samples have been transferred, can somebody restart the tests?

**Comment by Dr15Jones** [comment]
please test

**Comment by gpetruc** [comment]
ping?

**Comment by monttj** [comment]
+1

**Comment by gpetruc** [comment]
@danduggan @deguio : this requires also a DQM signature (if it didn't at the beginning when it was created, so maybe you had missed it)

File: DQMOffline/JetMET/python/jetMETDQMOfflineSource_cff.py [hunk 1]
```diff
@@ -61,6 +67,11 @@
     dqmAk4PFCHSL1FastL2L3ResidualCorrector
 )
 
+dqmAk4PFCHSL1FastL2L3Corrector = ak4PFCHSL1FastL2L3Corrector.clone()
```
Comment (deguio) [inline DQMOffline/JetMET/python/jetMETDQMOfflineSource_cff.py]: ciao @gpetruc 
aren't these lines duplicated? see line 50

Comment (arizzi) [inline DQMOffline/JetMET/python/jetMETDQMOfflineSource_cff.py]: @mmarionncern  can you please comment?

Comment (mmarionncern) [inline DQMOffline/JetMET/python/jetMETDQMOfflineSource_cff.py]: @deguio @arizzi  Those lines are complementary. One contains the residuals JEC and the other not.

**Comment by arizzi** [comment]
@deguio can you please sign then?

**Comment by monttj** [comment]
+1

**Comment by deguio** [comment]
+1

**Comment by deguio** [comment]
please test

**Comment by davidlange6** [comment]
+1

**Pull request merged**